src/analysis/utils/disk-utils.ts

```diff
@@ -59,10 +59,11 @@ export const defaultGceBootDiskSize = 120;
 export const defaultGcePersistentDiskSize = 50;
 export const defaultPersistentDiskType = googlePdTypes.standard;
 export const getCurrentAttachedDataDisk = (
-  app: App,
+  app: App | undefined,
   appDataDisks: PersistentDisk[]
 ): DecoratedPersistentDisk | undefined => {
-  const currentDisk: PersistentDisk | undefined = _.find({ name: app?.diskName }, appDataDisks);
+  const currentDisk: PersistentDisk | undefined =
+    app?.diskName === null ? undefined : appDataDisks.find(({ name }) => app?.diskName === name);
```

Comment:
If `app` is undefined, `app?.diskName` will be undefined, not null. That would mean that the `appDataDisks.find(...)` path would get run. That might be fine, if disks are guaranteed to have a name, then looking for one with an undefined name wouldn't return a result. That's what the current behavior is if `app` is undefined.

Alternatively, it might be worth loosening the condition to check if `app?.diskName` is falsy instead of strictly checking for null.

```suggestion
    !app?.diskName ? undefined : appDataDisks.find(({ name }) => app.diskName === name);
```

Reply:
I see, I am still new to the lodash syntax. Basically an app will always have a diskName, but it might be null, and in that case i want to return `undefined` for the persistent disk, which I think your suggested syntax is doing. Thanks for the explanation and pointers!

Reply:
The `?.` in `app?.diskName` is the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and is part of JavaScript, not Lodash.

`app?.diskName` is effectively `app === undefined || app === null ? undefined : app.diskName`.